File: JAVA/Leccion11/Ventas/src/ar/com/codesystem/ventas/Producto.java

package ar.com.codesystem.ventas;

public class Producto {
    // Atributos de la clase
    private int idProducto;
    private String nombre;
    private double precio;
    private static int contadorProductos;
    
    // Constructor vacío
    private Producto(){
        this.idProducto = ++Producto.contadorProductos;
    }
    // Constructor
    public Producto(String nombre, double precio){
        this(); // LLamamos al constructor vacío
        this.nombre = nombre;
        this.precio = precio;
    }
    
    //Getters and Setters
    public int getIdProducto() {
        return this.idProducto;
    }

    public String getNombre() {
        return this.nombre;
    }
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPrecio() {
        return this.precio;
    }
    public void setPrecio(double precio) {
        this.precio = precio;
    }

    // toString
    @Override
    public String toString() {
        return "Producto{" + "idProducto=" + idProducto + ", nombre=" + nombre + ", precio=" + precio + '}';
    }   
}

File: 2do Semestre/PYTHON/Leccion11/Producto.py
class Producto:
    contador_productos = 0 #Variable de clase

    def __init__(self, nombre, precio): 
        Producto.contador_productos += 1 #Aumento para la variable de clase 
        self._id_producto = Producto.contador_productos #Asignacion desde la variable de clase 
        self._nombre = nombre 
        self._precio = precio 

#Metodos setters and getters 
    @property
    def nombre(self):
        return self._nombre
    
    @nombre.setter
    def nombre(self, nombre):
        self._nombre = nombre

    @property
    def precio(self):
        return self._precio
    
    @precio.setter
    def precio(self, precio):
        self._precio = precio

    #Sobre escribimos el metodo str 
    def __str__(self):
        return f'Id Producto: {self._id_producto}, Nombre: {self._nombre}, Precio: {self._precio}'
    
if __name__ == '__main__': 
    producto1 = Producto('Remera', 150.00)
    print(producto1)
    producto2 = Producto('Short', 100.00)
    print(producto2)